bangc-ops/mlu_op.h

```diff
@@ -3776,13 +3776,266 @@ mluOpStatus_t MLUOP_WIN_API mluOpFill(
     mluOpHandle_t handle, const mluOpPointerMode_t pointer_mode,
     const void *value, const mluOpTensorDescriptor_t output_desc, void *output);
 
+
+// Group:RoiawarePool3d
+/*!
+ * @brief Returns in \b workspace_size the size of the MLU memory that is used as an extra
+ * workspace to optimize the RoiawarePool3dForward operation.
+ *
+ * The size of extra workspace is based on the given information of the RoiawarePool3dForward
+ * operation, including the input tensor descriptors \b pts_desc.
+ *
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and queues in the
+ * RoiawarePool3dForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] rois_desc
+ * Descriptor of rois, containing the dimension and layout of the rois tensor.
+ * @param[in] pts_desc
+ * Descriptor of pts, containing the dimension and layout of the pts tensor.
+ * @param[in] pts_feature_desc
+ * Descriptor of pts_feature, containing the dimension and layout of the pts_feature tensor.
+ * @param[out] workspace_size
+ * Pointer to the returned size of the extra workspace in bytes that is used in the
+ * RoiawarePool3dForward operation.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM
+ *
+ * @par Scale Limitation
+ * - None.
+ *
+ * @par Requirements
+ * - None.
+ *
+ * @par Example
+ * - None.
+ *
+ * @par Reference
+ * - None.
+ */
+mluOpStatus_t MLUOP_WIN_API
+mluOpGetRoiawarePool3dForwardWorkspaceSize(mluOpHandle_t handle,
+                                           const mluOpTensorDescriptor_t rois_desc,
+                                           const mluOpTensorDescriptor_t pts_desc,
+                                           const mluOpTensorDescriptor_t pts_feature_desc,
+                                           size_t *workspace_size);
+
+// Group:RoiawarePool3d
+/*!
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRoiawarePool3dForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] pool_method
+ * Pooling method of Roiaware, 0 is 'maxpool', 1 is 'avgpool'. Default: 0.
+ * @param[in] boxes_num
+ * An integer value which is the number of the rois.
+ * @param[in] pts_num
+ * An integer value which is the number of the pts.
+ * @param[in] channels
+ * An integer value which is the number of the pts feature of channels.
+ * @param[in] rois_desc
+ * Descriptor of rois tensor, containing the dimension and layout of the rois tensor.
```

Comment:
The descriptor of rois tensor, which contains the dimension and layout of the rois tensor.